backend/pkg/api/instances.go

```
@@ -101,6 +102,64 @@ type InstancesQueryParams struct {
 	Version       string `json:"version"`
 	Page          uint64 `json:"page"`
 	PerPage       uint64 `json:"perpage"`
+	SortFilter    string `json:"sort_filter"`
+	SortOrder     string `json:"sort_order"`
+}
+
+const (
+	id int = iota
+	ip
+	lastCheckForUpdates
+)
+
+var sortFilterMap = map[int]string{
+	id:                  "id",
+	ip:                  "ip",
+	lastCheckForUpdates: "last_check_for_updates",
+}
+
+type sortOrder int
+
+const (
+	sortOrderAsc sortOrder = iota
+	sortOrderDesc
+)
+
+func sortOrderFromString(str string) (sortOrder, error) {
+	val, err := strconv.Atoi(str)
+
+	return sortOrder(val), err
```

Here the returned value can be any `int` and there is a chance the value will not match both the cases in

```go
               if sortOrder == sortOrderAsc {
			instancesQuery = instancesQuery.Order(goqu.I(sortFilter).Asc().NullsLast())
		} else if sortOrder == sortOrderDesc {
			instancesQuery = instancesQuery.Order(goqu.I(sortFilter).Desc().NullsLast())
		}
```

---

Is there a need to return an error? In this case, the sortOrder will default to `sortOrderAsc`.